App.Stocks/Interfaces/ICompaniesRepository.cs

```diff
@@ -0,0 +1,14 @@
+﻿using App.Stocks.ModelsView;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace App.Stocks.Interfaces
+{
+	public interface ICompaniesRepository
+	{
+		IQueryable<Company> AllCompanies();
```

Comment:
if you are using async/await in managers, Repository is the part which should actually be async. Please, change interfaces of all repositories to return tasks